shared/src/main/scala/mlscript/ucs/Desugarer.scala

```diff
@@ -785,9 +786,39 @@ class Desugarer extends TypeDefs { self: Typer =>
       xs match {
         case MutCase.Constructor(className -> fields, cases) :: next =>
           printlnUCS(s"• Constructor pattern: $className(${fields.iterator.map(x => s"${x._1} -> ${x._2}").mkString(", ")})")
-          // TODO: expand bindings here
           val consequent = rec(cases)(defs ++ fields.iterator.map(_._2))
-          Case(className, mkLetFromFields(scrutinee, fields.toList, consequent), rec2(next))
+          val unapplyMtd = ctx.get(className.name) match {
+            case S(CompletedTypeInfo(cls: TypedNuCls)) => cls.td.genUnapply
+            case S(ti: DelayedTypeInfo) => ti.decl.genUnapply
+            case _ => N
```

Comment:
Enumerate what this is matching.

Reply:
Added comments for matching branches.

Reply:
You didn't enumerate anything... I've already told you in situations like these to avoid `case _ =>` and prefer `case _: OtherA | _: OtherB | ... =>`.

Reply:
Oh, I misunderstood that... I replaced the `case _ =>` branch.